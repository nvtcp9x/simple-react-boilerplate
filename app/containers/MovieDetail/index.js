// @flow

import * as React from 'react';
import { Row } from 'react-bootstrap'

import { StyledGrid } from '../../../utils/commonStyle'
import Poster from '../../components/Poster';
import Trailers from '../Trailers'
import Casts from '../Casts';
import Recommendations from '../Recommendations';
import Header from '../Header'

type Props = {
  match: {
    params: Object
  },
}

class MovieDetail extends React.Component<Props> {
  render() {
    const { id } = this.props.match.params;

    return (
      <StyledGrid>
        <Header />
        <Poster id={id} />
        <Row>
          <h3>Trailers</h3>
          <Trailers id={id} />
        </Row>
        <Row>
          <h3>Casts</h3>
          <Casts id={id} />
        </Row>
        <Row>
          <h3>Recommendations</h3>
          <Recommendations id={id} />
        </Row>
      </StyledGrid>
    )
  }
}


export default MovieDetail
