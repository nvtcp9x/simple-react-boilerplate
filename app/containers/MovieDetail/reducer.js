import { FETCH_MOVIE, FETCH_MOVIE_SUCCESS, FETCH_MOVIE_FAILURE } from './constants';

const initialState = {
  isFetching: false,
  item: {},
  error: {},
};

const movieDetail = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_MOVIE:
      return {
        ...state,
        isFetching: true,
      }
    case FETCH_MOVIE_SUCCESS:
      return {
        ...state,
        isFetching: true,
        item: action.data,
      }
    case FETCH_MOVIE_FAILURE:
      return {
        ...state,
        isFetching: true,
        error: action.error,
      }
    default:
      return state;
  }
}

export default movieDetail
