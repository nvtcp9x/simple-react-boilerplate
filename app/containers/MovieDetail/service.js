import axios from 'axios';
import { URL_DETAIL } from './constants';
import { API_KEY } from '../../../utils/constants';

export function fetchMovieDetail(id) {
  const urlMovie = `${URL_DETAIL}${id}?api_key=${API_KEY}`;
  return axios.get(urlMovie);
}
