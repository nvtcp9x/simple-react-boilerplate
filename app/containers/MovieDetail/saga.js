import { takeEvery, put, call } from 'redux-saga/effects';

import { FETCH_MOVIE } from './constants';
import { fetchMovieDetail } from './service';
import { fetchMovieSuccess, fetchMovieFail } from './actions';

export function* fetchMovie(action) {
  try {
    const { id } = action;
    const response = yield call(fetchMovieDetail, id);
    yield put(fetchMovieSuccess(response.data));
  } catch (err) {
    yield put(fetchMovieFail(err));
  }
}

export function* watchFetchMovie() {
  yield takeEvery(FETCH_MOVIE, fetchMovie)
}
