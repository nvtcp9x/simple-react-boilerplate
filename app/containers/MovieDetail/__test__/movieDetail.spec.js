import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import MovieDetail from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('MovieDetail component', () => {
  const mockProps = {
    match: {
      params: {
        id: 1,
      },
    },
  };

  const component = shallow(<MovieDetail {...mockProps} />);

  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
