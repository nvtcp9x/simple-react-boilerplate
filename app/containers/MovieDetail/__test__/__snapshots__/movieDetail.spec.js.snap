// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetail component should match snapshot 1`] = `
<Styled(Grid)>
  <withRouter(Connect(Header)) />
  <Extended
    id={1}
  />
  <Row
    bsClass="row"
    componentClass="div"
  >
    <h3>
      Trailers
    </h3>
    <Extended
      id={1}
    />
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <h3>
      Casts
    </h3>
    <Extended
      id={1}
    />
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <h3>
      Recommendations
    </h3>
    <Extended
      id={1}
    />
  </Row>
</Styled(Grid)>
`;
